Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2409,7 +2409,9 @@ GraphicsContext* CanvasRenderingContext2DBase::existingDrawingContext() const
 {
     if (!canvasBase().hasCreatedImageBuffer())
         return nullptr;
-    return drawingContext();
+    if (auto* buffer = canvasBase().buffer())
```

Comment:
I think this is a bit misleading. The expectation is really that:
- drawingContext() will create drawing buffer and return the drawing context
- existingDrawingContext() will return what drawingContext() would return, except null if it's not already created.

Could we instead:
```
bool CanvasRenderingContext2DBase::isAccelerated() const
{
    if (!canvasBase().hasCreatedImageBuffer())
        return false;
   auto* buffer = canvasBase().buffer();
   if (!buffer)
      return false;
   return buffer.renderingMode() == RenderingMode:::Accelerated;
}
```